libs/deck-layers/src/layers/vessel/VesselLayer.ts

```diff
@@ -1,108 +1,158 @@
 import { DataFilterExtension } from '@deck.gl/extensions'
-import { CompositeLayer, Layer, LayersList, LayerProps, Color } from '@deck.gl/core/typed'
+import {
+  CompositeLayer,
+  Layer,
+  LayersList,
+  LayerProps,
+  Color,
+  FilterContext,
+} from '@deck.gl/core/typed'
 // Layers
-import { ApiEvent, EventTypes, EventVessel, Segment } from '@globalfishingwatch/api-types'
+import {
+  ApiEvent,
+  EventTypes,
+  EventVessel,
+  ResourceStatus,
+  Segment,
+} from '@globalfishingwatch/api-types'
 import { parquetLoader } from '../../loaders/vessels/trackLoader'
 import { VesselDeckLayersEventData, vesselEventsLoader } from '../../loaders/vessels/eventsLoader'
 import { VesselDeckLayersEvent } from '../../layer-composer/types/vessel'
 import { deckToHexColor } from '../../utils/colors'
 import { EVENTS_COLORS, VesselEventsLayer, _VesselEventsLayerProps } from './VesselEventsLayer'
 import { VesselTrackLayer, _VesselTrackLayerProps } from './VesselTrackLayer'
 
+export const TRACK_LAYER_TYPE = 'track'
+export type VesselDataStatus = {
```

Comment:
💯 Nice way of aligning layers data state with general resources management on the platform